sound/soc/codecs/rt711-sdca-sdw.c

```diff
@@ -144,16 +144,16 @@ static int rt711_sdca_update_status(struct sdw_slave *slave,
 
 	if (status == SDW_SLAVE_ATTACHED) {
 		if (rt711->hs_jack) {
-			regcache_cache_only(rt711->regmap, false);
-			regcache_cache_bypass(rt711->regmap, true);
-			regmap_write(rt711->regmap, SDW_SCP_SDCA_INTMASK1,
+			/*
+			 * Due to the SCP_SDCA_INTMASK will be cleared by any reset, and then
+			 * if the device attached again, we will need to set the setting back.
+			 * It could avoid losing the jack detection interrupt.
+			 * This also could sync with the cache value as the rt711_sdca_jack_init set.
+			 */
+			sdw_write_no_pm(rt711->slave, SDW_SCP_SDCA_INTMASK1,
 				SDW_SCP_SDCA_INTMASK_SDCA_0);
-			regmap_write(rt711->regmap, SDW_SCP_SDCA_INTMASK2,
+			sdw_write_no_pm(rt711->slave, SDW_SCP_SDCA_INTMASK2,
```

Comment:
We should be really careful to access a register by regmap and the IO function directly. The code looks fine to me since we set the same value here as rt711_sdca_jack_init(), and we don't read it. But I will wait for @plbossart 's comment.

Reply:
indeed we should not have the INTMASK1/2 registers in regmap, so what's missing is the removal of those registers from the regmap definition. 
I think I added this in another PR some time back?

Reply:
#2622